### PR TITLE
TravisCI: More robust parsing of TravisCI log

### PR DESCRIPTION
**Pull request by tammoippen**
Solves #392.
This PR makes the parsing of the TravisCI more robust and allows to order the different steps of the `build.sh` in any order (for the parsing). It is also possible to remove different steps completely and the parser will not break. A summary of the changes:

In `build.sh`:
* Every block is surrounded by tags like 

        ======= Extract changed files start =======
        ...
        ======= Extract changed files end =======
* Output is reduced:
  - Comment [`set -x`](http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_02_03.html#sect_02_03_02) out, as we do not need to debug the travis log regularly. One can uncomment it, if needed.
  - Remove verbose flag of extracting `clang-format`.
* Make static analysis optional with env. variable `xSTATIC_ANALYSIS`.
* Fix [regex bug](https://travis-ci.org/nest/nest-simulator/jobs/136773603#L4188) (see line 4188) for `if [[ $f =~ $EXAMPLE_DIRS ]]`. I think, the TravisCI environment does not allow `if [[ ... ]]`.
* Add output that tells if results are not uploaded, if you are working on your fork.

In `extras/parse_travis_log.py`:
  * No state on reading/parsing the log. Every function opens the file itself and completely parses it for its issues. The actual parsing happens only within the starting/ending tags relevant for this function.
  * Separate `vera++`, `cppcheck` and `clang-format` parsing into standalone functions.
  * Add `print_...` functions for nicer output.
  * Cleanup "`main`" function.

I propose @heplesser and @gtrensch as reviewers.

### CODE REVIEW

**Comment by gtrensch** [comment]
:+1: Everything works fine!  Just one remark from my side: There is a comment in parse_travis_log.py referring to the dependencies on build.sh. A similar comment in build.sh would be good.

File: build.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 set -e
-set -x
+#set -x
```
Comment (heplesser) [inline build.sh]: Why commented out instead of deleted?

File: extras/parse_travis_log.py [hunk 2]
```diff
@@ -49,81 +53,88 @@ def process_installcheck(f):
...
-                           line.strip().split('=')[1].split(' ')), line)
-
-
-def process_vera(f, filename):
+    in_changed_files_section = False
+    for line in open(f, "r"):
+        if (not in_changed_files_section and
+                line.strip() == "======= Extract changed files start ======="):
+            in_changed_files_section = True
+        if in_changed_files_section and line.startswith('file_names='):
```
Comment (heplesser) [inline extras/parse_travis_log.py]: Shouldn't this be `line.strip().startswith('file_names=')` to be on the safe side? Applies also to all other uses of `line.startswith()`.

File: build.sh (hunk 1 above)
Comment (tammoippen) [inline build.sh]: @heplesser I documented the 'out-commenting'. When debugging the script, the flag is very [informative](http://julio.meroh.net/2010/01/set-e-and-set-x.html).

**Comment by tammoippen** [comment]
@gtrensch I added the comment in 500f192. Thanks.

File: extras/parse_travis_log.py (hunk 2 above)
Comment (tammoippen) [inline extras/parse_travis_log.py]: @heplesser I added the `strip()`'s in 4033e3e.

**Comment by tammoippen** [comment]
@heplesser in 749d929 I introduced some intentional failures that should trigger 'vera, clang-format, cppcheck, pep8, building, testsuite'. I will remove it afterwards.

**Comment by tammoippen** [comment]
See this [build](https://travis-ci.org/nest/nest-simulator/jobs/138848521) for forced errors.

**Comment by heplesser** [comment]
:+1: and merging.

**Pull request merged**